src/compressAlignment.cpp

```diff
@@ -57,10 +70,145 @@ void shasta::compress(const Alignment& alignment, string& s)
+        }
 
+        ordinal0 += (streakSize-1);
+        ordinal1 += (streakSize-1);
+    }
+}
 
-void shasta::decompress(span<const char>, Alignment)
-{
-    SHASTA_ASSERT(0);
+uint8_t shasta::compressAlignment::extractFormatIdentifier(const char c) {
+    if ((c & Format0::ID_MASK) == Format0::ID) {
+        return Format0::ID;
+    } else if ((c & Format1::ID_MASK) == Format1::ID) {
+        return Format1::ID;
+    } else if ((c & Format2::ID_MASK) == Format2::ID) {
+        return Format2::ID;
+    } else if ((c & Format3::ID_MASK) == Format3::ID) {
+        return Format3::ID;
+    } else if ((c & Format4::ID_MASK) == Format4::ID) {
+        return Format4::ID;
+    } else {
+        string errorStr("Unsupported format in byte - ");
+        for (uint8_t i = 0; i < sizeof(c) * 8; i++) {                                                          
+            errorStr.append(1, !!((c << i) & 0x80) ? '1': '0');                                                                     
+        }
+        errorStr.append(".");
+        throw std::runtime_error(errorStr);
+    }
 }
 
+void shasta::testAlignmentCompression() {
+    vector< array<uint32_t, 2> > ordinals {
+        {300, 200}, // First streak begins here (Format2)
+        {301, 201},
+        {302, 202},
+        {305, 206}, // Second streak begins here (Format1)
+        {306, 207},
+        {320, 250}, // Third streak begins here (Format2)
+        {321, 251},
+        {322, 252},
+        {323, 253},
+        {325, 255}, // Fourth streak begins here (Format0)
+        {326, 256},
+        {350, 257}, // Fifth streak begins here (Format2)
+        {351, 258},
+        {352, 259},
+        {353, 260},
+        {354, 261},
+        {1000, 400}, // Sixth streak begins here (Format3)
+        {1001, 401},
+        {1002, 402},
+        {600000, 500000}, // Seventh streak begins here (Format4)
+        {600001, 500001},
+        {500000, 500005}, // Eighth streak (negative skip0) (Format3)
+        {500001, 500006},
+    };
+  
+    Alignment alignment;
+    alignment.ordinals = ordinals;
+
+    // cout << "---Original alignment-----" << endl;
+    // for(const auto& x: alignment.ordinals) {
+    //     cout << x[0] << ", " << x[1] << endl;
+    // }
+    // cout << "--------------------------" << endl;
+
+    string compressed;
+    shasta::compress(alignment, compressed);
+
+    cout << "Uncompressed size = " << ordinals.size() * sizeof(uint32_t) * 2 << " bytes." << endl;
+    cout << "Compressed size = " << compressed.size() << " bytes." << endl;
+
+    Alignment decompressedAlignment;
+    shasta::decompress(compressed, decompressedAlignment);
+
+    // cout << "---Decompressed Alignment -----" << endl;
+    // for(const auto& x: decompressedAlignment.ordinals) {
+    //     cout << x[0] << ", " << x[1] << endl;
+    // }
+    // cout << "-------------------------------" << endl;
+
+    SHASTA_ASSERT(alignment.ordinals.size() == decompressedAlignment.ordinals.size());
```

Comment:
These four lines can be written more compactly as 

`SHASTA_ASSERT(alignment.ordinals == decompressedAlignment.ordinals);`

Reply:
Good point. Will change.